### ISSUE TITLE
Use defined tableName (instead of collectionName) in the junctionTable creation for STI

### ISSUE THREAD

**Issue by Zaggen** [balderdashy/waterline-schema#36]
If we define a model/collection `User` where we define a tableName and have lets say a one way association to another model, there is no way to have the created junction table with the tableName. Now, **why on earth is this useful?** If we want to ***emulate STI*** (Single table inheritance) like in ruby, is a must (at least in my opinion). Btw if this functionality actually exist, please let me know and maybe we can update the docs (sails docs mostly).

Asumming i'm using a module to create mixins, like [`def-type`](https://www.npmjs.com/package/def-type), but it can be lodash.extend or anything similar.

***profileTraits.js*** (This object will be merged into others) 
```js
Var profileTraits =  {
   tableName: 'profile'
  attributes: {
    name: {type: 'string'}
    types: {collection: 'ProfileType', useTableName: true} // By using useTableName the same junctionTable will be used by all models that inherit from this (because we used the tableName instead of ModelName)
  }
};
```
***Profile.js*** (Our default profile model)
```js
def = require('def-type');
profileTraits = require('..../profileTraits.js') 
Var Profile = def.Object({
   merges: [profileTraits] // pretty straight forward, but this merges the properties of profileTraits into this obj
});
```

Now lets say we have a more specialized one
***Artist.js***
```js
def = require('def-type');
profileTraits = require('..../profileTraits.js')
Var Artist = def.Object({
   merges: [profileTraits],
   attributes: 
     artwork: {collection: 'Artwork', via: 'artists'}, // A junction table will be created using modelName as part of it
     events: {collection: 'Event', via: 'artists'}  // A junction table will be created using modelName as part of it
});
```

Now we can create custom finders in our models, but that's a different story, i need this functionality precisely to do STI with the option to use the default of letting waterline create junctionTables based on the modelName when those are not related to parent.***I've already created this*** functionality for my own use case (i've only test it on one way associations) but i really think waterline users can benefit a lot from it, since STI yields a lot of power (IMHO) and its a necessary step (Not the only step though), and i'm willing to do a full pull request once i have some time to create tests and check everything, but i would like to know what you guys think (@particlebanana, @mikermcneil, @dmarcelino, @waterlineUsers ) and if you'll accept this feature if i create the PR.

Have a nice day :)

**Comment by sailsbot** [comment]
@Zaggen: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by Zaggen** [comment]
Still waiting for a reply here. I think that just as we have the option to set the tableName, we should be allow to use that tableName in the junctionTable name, instead of the model name, and this allows this. Now i'm not sure if `useTableName: true` is the best api, any ideas are welcomed.

**Comment by sailsbot** [comment]
@Zaggen,@sailsbot: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by Zaggen** [comment]
Still waiting...

**Comment by particlebanana** [comment]
@Zaggen if you are looking to customize the generated `junctionTables` you can just create model's for them instead of relying on the automatically generated ones.

**Comment by Zaggen** [comment]
@particlebanana Yes i'm aware of that, but i feel we should have this level of control, if i'm already specifying a custom tablename, i wan't my junction tables to be generated using that tablename if i want to. And like i said, i could help to implement STI in the future, if i have a junction model, i can only reference the mainModel and the other i want to associate, following the example above, i could only associate 

`Profile` to `Type` in the custom model and not `Artists` to `Type`
Profile Model
```
Var Profile = {
   attributes: {
     types:{
      collection: 'type',
      via: 'profile',
      through: 'profiletypes'
    }
 }
}
```

Artists model
```
Var Artist = def.Object({
   extends: Profile
})
```
Through Model
```
var ProfileTypes = 
  attributes: {
    profile:{
      model:'profile'
    },
    types: {
      collection: 'type'
    }
  }
```

This throws an error when loading the Artist model, since its not the actual profile model, its just inheriting. 

pst: I coding from the back of my head, i'm not sure of the syntax but i remeber it throwed that kind of error, so it did't help me to emulate STI, but what i'm proposing (that i've already implemented) did.

pst: Of course there is more involved to actually implement sti on waterline, but it could be a powerful adition that orms like rails activeRecord provides.

I hope i made my point this time, or if its actually posible and i'm twisting things around for now reason let me know

**Comment by sailsbot** [comment]
@Zaggen,@sailsbot,@particlebanana: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!
